### ISSUE TITLE
Empty Style tag breaks .vue file highlighting

### ISSUE THREAD

**Issue by Thevetat** [nvim-treesitter/nvim-treesitter#5490]
### Describe the bug

In a vue file, if i add this at the bottom:

```
<style lang="scss" scoped> </style>
```

It breaks the treesitter highlighting.  The playground reports a whole new block of errors.

Removing the above line will allow the auto tag closing and highlghting to work properly.

### To Reproduce

See description

### Output of `:checkhealth nvim-treesitter`

```markdown
Installation ~
- OK `tree-sitter` found 0.20.8 (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v18.16.1 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: Apple clang version 14.0.3 (clang-1403.0.22.14.1)
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "arm64",
  release = "22.6.0",
  sysname = "Darwin",
  version = "Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:53 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6020"
} ~
```

### Output of `nvim --version`

```text
NVIM v0.10.0-dev-653+gfbeef0d4e-Homebrew
Build type: Release
LuaJIT 2.1.0-beta3
Run "nvim -V1 -v" for more info
```

**Comment by clason** [comment]
> The playground reports a whole new block of errors.

This indicates a parser issue; please report the problem there.

**Comment by amaanq** [comment]
you're either not posting a real snippet that's a minimum reproducer or you're using some plugin that breaks vue highlighting - the output of parsing `<style lang="scss" scoped> </style>` with tree-sitter-vue is totally normal

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/29718261/0c47c276-0239-4d01-8b89-5e9ce0229225)

**Comment by Thevetat** [comment]
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/126312031/2fc79ff6-5a59-4e5d-8187-f88002fff074)

Apologize, that is what I am seeing.

**Comment by amaanq** [comment]
It would've been nice if you posted the entire snippet, that's needed for reproducing, and yeah, this belongs upstream at tree-sitter-vue if that is in fact valid syntax

**Comment by lucario387** [comment]
The actual problem here is the version of Neovim he's using doesn't have the patch for empty range injection handling that lewis fixed last month @amaanq 

Please update your neovim nightly. The fix is already backported to 0.9.2 as well

**Comment by Thevetat** [comment]
Really appreciate your comment lucario.  That did indeed fix it dropping back to 0.9.2.
